### PR TITLE
Analytics

### PR DESCRIPTION
**Pull request by CityVibes**
Placed initialization into BaseActivity as we do not use Application class and which activity is launched first might eventually change.
Used "screen" method to track screens, "identify" for logging in, and "track" for everything else.

### CODE REVIEW

File: devicesetup/build.gradle [hunk 1]
```diff
@@ -1,10 +1,11 @@
 apply plugin: 'com.android.library'
 apply plugin: 'me.tatarka.retrolambda'
 
-
 // This is the library version used when deploying the artifact
 version = '0.4.0'
 
+def analytics = "false"
```
Comment (jensck) [inline devicesetup/build.gradle]: Are you planning to just change these values manually on your local build and then not commit the changes?

**Comment by jensck** [comment]
Got a couple things here to consider, but nothing that blocks approval.  Merge when you're ready.

**Comment by CityVibes** [comment]
Changed the way analytics are enabled and used.

**Comment by CityVibes** [comment]
@jensck Made changes as per your suggestions.

File: devicesetup/src/main/java/io/particle/android/sdk/utils/SEGAnalytics.java [hunk 2]
```diff
@@ -9,11 +11,14 @@
 /**
  * Created by Julius.
  */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
```
Comment (jensck) [inline devicesetup/src/main/java/io/particle/android/sdk/utils/SEGAnalytics.java]: 👍

**Pull request merged**